docs/rfc/0002-processor-interface.md

```diff
@@ -0,0 +1,110 @@
+# Processor Interface RFC
+
+- Contribution Name: Processor Interface/Plugin Framework
+- Implementation Owner: Stephen Akiki
+
+## Problem Statement
+
+Users of the Indexer each have their own needs when it comes to the data gathered by the Indexer. Whether it is
+extracting certain fields of information or transforming it entirely, the data gathered is something that every user
+consumes in a different way.
+
+Currently, users must "drink from a firehouse" of information in that all information is presented at all times. No
+custom "pre-processing" of the data can be done so no derivative information can be created. This means that the user
+must save all the data and only after that is accomplished can they use the information. This can result in an
+incredible amount of "wasted" storage if the user only needs to save a derived value and not all underlying data.
+
+Additionally, there might be various transformations that are useful enough that one would wish to "pipe" them from one
+to another, similar to how commands are piped in Linux. This would allow for modules to be saved and re-used across
+Indexer deployments.
+
+## Proposal
+
+### Plugin Interface
+
+A sample interface and pipeline effort is shown below:
+
+```go
+package processor
+
+// Config will act as a placeholder for now. It will end up providing an interface for
+// serialization/deserialization of config files.
+// Derived types will provide plugin-specific data fields.
+type Config string
+
+type MetaData string
+
+type Processable interface {
+    // Metadata associated with each Processor.
+    Metadata() MetaData
+
+    // Config returns the configuration options used to create an Processor.
+    // Initialized during `Connect`, it will return nil until the Exporter has been Connected.
+    Config() Config
+
+    // Init will be called during initialization, before block data starts going through the pipeline.
+    // Typically used for things like initializating network connections.
+    // The Config passed to Connect will contain the Unmarhsalled config file specific to this plugin.
+    Init(cfg Config) error
+
+    // Close will be called during termination of the Indexer process.
+    // There is no guarantee that plugin lifecycle hooks will be invoked in any specific order in relation to one another.
+    Close() error
+
+    // Process will be called with provided optional inputs.  It is up to the plugin to check that required inputs are provided.
+    Process(input BlockData) (BlockData, error)
+}
+```
+
+### Interface Description
+
+#### Metadata and Configuration
+
+The `Metadata()` and `Config()` functions are used to get the supplied information of the interface.  The information will be returned as a string, which can be used along-side conduit tooling to de-serialize into appropriate objects.
+
+Plugin Metadata shall adhere to the Conduit `PluginMetadata` interface and return the appropriate type depending on the defined plugin.  
```

Comment:
What is this?

Reply:
Essentially this: https://github.com/algorand/indexer/pull/1114/files#diff-f935b82f49f654b4a0a7256dae85ea3d50213e25c317dc7ddf1abb2c9a090730

I'm not sure how to describe it well.

Reply:
I would just not really address the Metadata stuff here... I think it's good to have in our interface, but it's mostly going to be useful for the framework in the end.

Reply:
Do you mean I should remove it from this rfc or say something like "refer to Conduit documentation for more info"

Reply:
Either way is fine in my opinion.